synapse/lib/stormtypes.py

```diff
@@ -8270,6 +8279,15 @@ async def _methUserAllowed(self, permname, gateiden=None, default=False):
         user = await self.runt.snap.core.auth.reqUser(self.valu)
         return user.allowed(perm, gateiden=gateiden, default=default)
 
+    async def getAllowedReason(self, permname, gateiden=None, default=False):
+        permname = await tostr(permname)
+        gateiden = await tostr(gateiden)
+        default = await tobool(default)
+
+        perm = tuple(permname.split('.'))
```

Comment:
This should probably be using `ruleFromText` to parse deny rules such as `!foo.bar.baz`